juniper/src/value.rs

```diff
@@ -22,7 +21,115 @@ pub enum Value {
     String(String),
     Boolean(bool),
     List(Vec<Value>),
-    Object(IndexMap<String, Value>),
+    Object(Object),
+}
+
+/// A Object value
+#[derive(Debug, PartialEq, Clone)]
+pub struct Object {
+    key_value_list: Vec<(String, Value)>,
+}
+
+impl Object {
+    /// Create a new Object value with a fixed number of
+    /// preallocated slots for field-value pairs
+    pub fn with_capacity(size: usize) -> Self {
+        Object {
+            key_value_list: Vec::with_capacity(size),
+        }
+    }
+
+    /// Add a new field with a value
+    ///
+    /// If there is already a field with the same name the old value
+    /// is returned
+    pub fn add_field<K>(&mut self, k: K, value: Value) -> Option<Value>
+    where
+        K: Into<String>,
+        for<'a> &'a str: PartialEq<K>,
+    {
+        if let Some(item) = self.key_value_list
+            .iter_mut()
+            .find(|(key, _)| (key as &str) == k)
+        {
+            return Some(::std::mem::replace(&mut item.1, value));
+        }
+        self.key_value_list.push((k.into(), value));
+        None
+    }
+
+    /// Check if the object already contains a field with the given name
+    pub fn contains_field<K>(&self, f: K) -> bool
+    where
+        for<'a> &'a str: PartialEq<K>,
+    {
+        self.key_value_list
+            .iter()
+            .any(|(key, _)| (key as &str) == f)
+    }
+
+
+    /// Get a iterator over all field value pairs
+    pub fn iter(&self) -> impl Iterator<Item = &(String, Value)> {
```

Comment:
Hmmm, looks like this uses impl trait, which didn't stabilize until [1.26](https://blog.rust-lang.org/2018/05/10/Rust-1.26.html). Supporting impl trait would mean we would have to bump our min rust version.